libraries/core_libs/consensus/src/dag/dag.cpp

```diff
@@ -418,6 +418,14 @@ void DagManager::addDagBlock(DagBlock const &blk, SharedTransactions &&trxs, boo
           LOG(log_dg_) << "Block already in DB: " << blk.getHash();
           return;
         }
+        const auto proposal_period = dag_blk_mgr_->getProposalPeriod(blk.getLevel());
+        assert(proposal_period.second);
+        if (proposal_period.first + FullNodeConfig::kDagExpiryPeriodLimit < period_) {
```

Comment:
same check could be part of `verifyBlock` as we do not need to verify block if it is too old

Reply:
I think it is better that it is here under the DagManager mutex. Otherwise there could be a race condition that when we verify block it is not old but when it is inserted to DAG period changed and it could be old.

Reply:
but it could be in both. This way we could save some CPU if we receive old blocks

Reply:
I have put it now in both